File: src/main/java/net/alexwells/roomery/mechanic/roomcard/RoomCardItem.java
package net.alexwells.roomery.mechanic.roomcard;

import net.alexwells.roomery.Roomery;
import net.minecraft.item.Item;

public class RoomCardItem extends Item {
    public static final String NAME = "room_card";

    public RoomCardItem() {
        super();

        setMaxStackSize(16);
        setRegistryName(Roomery.MOD_ID, NAME);
        setTranslationKey(getRegistryName().toString());
        setCreativeTab(Roomery.instance().creativeTab());
    }
}

File: src/main/java/net/alexwells/roomery/mechanic/roomcard/RoomCardItem.kt
package net.alexwells.roomery.mechanic.roomcard

import net.alexwells.roomery.Roomery
import net.alexwells.roomery.util.createResource
import net.minecraft.item.Item
import net.minecraft.item.ItemStack

object RoomCardItem : Item(
    Item.Properties()
        .maxStackSize(16)
        .group(Roomery.itemGroup)
) {
    const val ID_TAG = "Id"

    init {
        registryName = createResource("room_card")
    }

    fun isValid(stack: ItemStack): Boolean = stack.item == this && stack.hasTag() && stack.tag!!.contains(ID_TAG)
    fun getId(stack: ItemStack): String? = if (isValid(stack)) stack.tag!!.getString(ID_TAG) else null
}
